proximal/examples/test_deconv_sv_psf.py

```diff
@@ -0,0 +1,320 @@
+    # redundant 2D convolution with zero weights.
+    data_term = px.weighted_sum_squares(im_blur - im_stack, weights)
+
+    im_grad = px.grad(x)
+    grad_term = mu * px.norm1(im_grad) + (1 - mu) * px.sum_squares(im_grad)
+    return data_term + grad_term  #+ px.nonneg(x)
+
+
+def FlickerAndRigaut2005(x,
+                         im,
+                         psf_modes,
+                         weights,
+                         mu=0.01,
+                         alpha=0.9,
+                         n_psf=15,
+                         using_blackbox=False):
+    ''' Weight, then convolve.
+
+        According to Eq. 22 of the review article (link below), PSF
+        interpolation technique preserves PSF symmetry and normalization with
+        minimal sacrifice of the computation complexity.
+         
+        Reference: Denis, Thiebaut, Soulez et al. 2015,
+        https://doi.org/10.1007/s11263-015-0817-x
+    '''
+    if using_blackbox:
+        blur_op = weighted_planewise_conv(psf_modes, weights, im.shape, n_psf)
+        im_blur = blur_op(x)
+    else:
+        im_blur = px.sum([
+            px.conv(psf_modes[i], px.mul_elemwise(weights[i], x))
+            for i in range(n_psf)
+        ])
+
+    # The weights are manually "absorbed" into the L2-norm, which has a fast
+    # elemwise proximal operator.
+    #
+    # TODO: Crop the im_stack and weight to double of image grid size; elminiate
+    # redundant 2D convolution with zero weights.
+    data_term = px.sum_squares(im_blur - im)
+
+    im_grad = px.grad(x)
+    grad_term = mu * alpha * n_psf * px.group_norm1(im_grad, group_dims=[
+        2
+    ]) + mu * (1 - alpha) * n_psf * px.sum_squares(im_grad)
+    return data_term + grad_term
+
+
+def draw(im, idx: int, title: str):
+    plt.subplot(3, 4, idx)
+    plt.imshow(im, cmap='gray')
+    plt.axis('off')
+    plt.title(title)
+
+
+def get_psf(n_psf: int, kwidth: int):
+    assert n_psf <= 4
+
+    psf_modes = np.empty((n_psf, kwidth, kwidth), dtype=np.float32)
+    psf_modes[0] = get_kernel(kwidth, 2)
+
+    for i in range(1, n_psf):
+        psf_modes[i] = np.rot90(psf_modes[0], k=i)
+
+    return psf_modes
+
+
+def get_mask(n_psf: int, imwidth: int, feather_edge: int = 100):
+    assert n_psf <= 4
+
+    mask = np.zeros((n_psf, imwidth, imwidth), dtype=np.float32)
+
+    W = int(imwidth / 2)
+    mask[0, :W, :W] = 1
+    mask[0] = gaussian_filter(mask[0], feather_edge)
+
+    for i in range(1, n_psf):
+        mask[i] = np.rot90(mask[0], k=i)
+
+    # Normalize the weig
+    mask[:] /= np.sum(mask, axis=0)[np.newaxis, ...]
+
+    return mask
+
+
+def simulate(im, psf_modes, mask, noise_level: float = 1e-2):
+    _x = px.Variable(im.shape)
+    blur_op = weighted_planewise_conv(psf_modes, mask, im.shape, n_psf)(_x)
+    im_sb = np.empty(im.shape, dtype=np.float32, order='F')
+    blur_op.forward([im], [im_sb])
+
+    im_sb += noise_level * im_sb.max() * np.random.randn(*im.shape)
+
+    return im_sb
```

Comment:
Given the high-resolution image as the ground truth, simulate the spatial variant blur, and then add Gaussian noise.